pkg/v1/cli/command/core/discovery_source.go

```diff
@@ -0,0 +1,297 @@
+		err = config.StoreClientConfig(cfg)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+var deleteDiscoverySourceCmd = &cobra.Command{
+	Use:   "delete [name]",
+	Short: "Delete a discovery source",
+	Args:  cobra.ExactArgs(1),
+	Example: `
+    # Delete a discovery source
+    tanzu plugin discovery delete standalone-oci`,
+	RunE: func(cmd *cobra.Command, args []string) (err error) {
+		discoveryName := args[0]
+		cfg, err := config.GetClientConfig()
+		if err != nil {
+			return err
+		}
+		if cfg.ClientOptions == nil || cfg.ClientOptions.CLI == nil {
+			return fmt.Errorf("discovery %q unknown", discoveryName)
+		}
+
+		newDiscoverySources, err := deleteDiscoverySource(cfg.ClientOptions.CLI.DiscoverySources, discoveryName)
+		if err != nil {
+			return err
+		}
+
+		cfg.ClientOptions.CLI.DiscoverySources = newDiscoverySources
+		err = config.StoreClientConfig(cfg)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+}
+
+func createDiscoverySource(dsType, dsName, uri string) (configv1alpha1.PluginDiscovery, error) {
+	pluginDiscoverySource := configv1alpha1.PluginDiscovery{}
+	if dsType == "" {
+		return pluginDiscoverySource, errors.New("discovery source type cannot be empty")
+	}
+	if dsName == "" {
+		return pluginDiscoverySource, errors.New("discovery source name cannot be empty")
+	}
+
+	switch strings.ToLower(dsType) {
+	case common.DiscoveryTypeLocal:
+		pluginDiscoverySource.Local = createLocalDiscoverySource(dsName, uri)
+	case common.DiscoveryTypeOCI:
+		pluginDiscoverySource.OCI = createOCIDiscoverySource(dsName, uri)
+	default:
+		return pluginDiscoverySource, errors.Errorf("discovery source type '%s' is not yet supported", dsType)
+	}
+	return pluginDiscoverySource, nil
+}
+
+func createLocalDiscoverySource(discoveryName, uri string) *configv1alpha1.LocalDiscovery {
+	return &configv1alpha1.LocalDiscovery{
+		Name: discoveryName,
+		Path: uri,
+	}
+}
+
+func createOCIDiscoverySource(discoveryName, uri string) *configv1alpha1.OCIDiscovery {
+	return &configv1alpha1.OCIDiscovery{
+		Name:  discoveryName,
+		Image: uri,
+	}
+}
+
+// checkDiscoveryName returns true if discovery name exists else return false
+func checkDiscoveryName(ds configv1alpha1.PluginDiscovery, dn string) bool {
+	return (ds.GCP != nil && ds.GCP.Name == dn) ||
+		(ds.Kubernetes != nil && ds.Kubernetes.Name == dn) ||
+		(ds.Local != nil && ds.Local.Name == dn) ||
+		(ds.REST != nil && ds.REST.Name == dn) ||
+		(ds.OCI != nil && ds.OCI.Name == dn)
+}
+
+func discoverySourceNameAndType(ds configv1alpha1.PluginDiscovery) (string, string) {
+	if ds.GCP != nil {
+		return ds.GCP.Name, common.DiscoveryTypeGCP
+	} else if ds.Kubernetes != nil {
+		return ds.Kubernetes.Name, common.DiscoveryTypeKubernetes
+	} else if ds.Local != nil {
+		return ds.Local.Name, common.DiscoveryTypeLocal
+	} else if ds.OCI != nil {
+		return ds.OCI.Name, common.DiscoveryTypeOCI
+	} else if ds.REST != nil {
+		return ds.REST.Name, common.DiscoveryTypeREST
+	} else {
```

Comment:
Thanks for extracting this as a separate function. 

nit: This would look clearer as a `switch` statement.